modules/users/server/controllers/users.profile.server.controller.js

```diff
@@ -1503,7 +1505,9 @@ exports.search = function (req, res, next) {
     .select(exports.userMiniProfileFields + ' -_id')
     .sort({ score: { $meta: 'textScore' } })
     // limit the amount of found users (config)
-    .limit(config.limits.userSearchLimit)
+    .limit(config.limits.userSearchPageLimit)
+    // skip to the page, automatically handles invalid page number
+    .skip((req.query.page-1)*config.limits.userSearchPageLimit)
```

Comment:
I'm still quite curious to understand the behavior of this thing.
- `req.query.page` is a string. can be parsed like `+req.query.page`.
- `req.query.page` can benefit from a validation (only positive integer allowed).

Reply:
`+req.query.page` evaluates to 1 when it is xxx, 0 or negative. Same with `parseInt(req.query.page)`
I don't really know what to do with that

Reply:
We actually have a pagination middleware helper; take a look at threads and messages apis.

Reply:
Here:

https://github.com/Trustroots/trustroots/blob/f6893632a696d56ecfe64696313ac025112e0169/modules/messages/server/models/thread.server.model.js#L39

https://github.com/Trustroots/trustroots/blob/master/modules/messages/server/controllers/messages.server.controller.js#L145

Reply:
Actually thinking further:

- [Mongoose Paginate](https://github.com/edwardhotchkiss/mongoose-paginate) module that we use isn't maintained anymore

- We still don't want to replicate Pagination headers, query validation etc etc in every API endpoint, so we'll keep using well maintained [Express.js paginate](https://github.com/expressjs/express-paginate)

- We can just do the actual limit/pagination just like you do in this PR (see also [this example](https://github.com/expressjs/express-paginate#example-with-mongoose-odm-see-example-2-for-sequelize-orm)) — we should just ensure we rely on Express pagination so that we don't have one API endpoint outside it while the rest are using it.

Reply:
So the express-paginate can take care of a validation for us, right? Or is there more in it?

Reply:
@petr-hajek It was probably the `express-paginate` which was doing weird things with our `req.query.page`, setting it to 1. (Remember the confusion when we talked about it in person?)

It is set up in `/config/lib/express.js` and was doing funny validation stuff without our knowledge.

Reply:
This is resolved with bba47c14d6b48b008d6b712.
We use `mongoose-paginate` with the knowledge that we use it.